src/wp-includes/class-wp-block.php

```diff
@@ -506,14 +536,24 @@ public function render( $options = array() ) {
 					if ( ! is_null( $pre_render ) ) {
 						$block_content .= $pre_render;
 					} else {
-						$source_block = $inner_block->parsed_block;
+						$source_block        = $inner_block->parsed_block;
+						$inner_block_context = $inner_block->context;
 
 						/** This filter is documented in wp-includes/blocks.php */
 						$inner_block->parsed_block = apply_filters( 'render_block_data', $inner_block->parsed_block, $source_block, $parent_block );
 
 						/** This filter is documented in wp-includes/blocks.php */
 						$inner_block->context = apply_filters( 'render_block_context', $inner_block->context, $inner_block->parsed_block, $parent_block );
 
+						if ( $inner_block->context !== $inner_block_context ) {
+							$inner_block->available_context = array_merge( $this->available_context, $inner_block->context );
```

Comment:
This line doesn't look right to me. If we want to update `$available_context` based on `$context`, this should happen within the `refresh_context_dependents()` because then `$available_context` is a dependent of `$context`.

Reply:
Thanks for the feedback. make sense. Updated in https://github.com/WordPress/wordpress-develop/pull/7522/commits/4096831e6f690e1bdab1b5fde7fc661f0c1994af